src/useComponentSelectorRule.ts

```diff
@@ -0,0 +1,41 @@
+import { sprintf } from 'sprintf-js';
+import { IRuleMetadata, RuleFailure } from 'tslint';
+import { AbstractRule } from 'tslint/lib/rules';
+import { SourceFile } from 'typescript';
+import { ComponentMetadata } from './angular/metadata';
+import { NgWalker } from './angular/ngWalker';
+
+export const getFailureMessage = (componentName: string): string => sprintf(Rule.FAILURE_STRING, componentName);
+
+export class Rule extends AbstractRule {
+  static readonly metadata: IRuleMetadata = {
+    description: 'Component selector must be declared.',
+    options: null,
+    optionsDescription: 'Not configurable.',
+    rationale: 'Omit the component selector makes debugging difficult.',
+    ruleName: 'use-component-selector',
+    type: 'maintainability',
+    typescriptOnly: true
+  };
+
+  static readonly FAILURE_STRING = 'The selector of the component "%s" is mandatory';
+
+  apply(sourceFile: SourceFile): RuleFailure[] {
+    return this.applyWithWalker(new UseComponentSelectorValidatorWalker(sourceFile, this.getOptions()));
+  }
+}
+
+export class UseComponentSelectorValidatorWalker extends NgWalker {
+  protected visitNgComponent(metadata: ComponentMetadata): void {
+    this.validateComponent(metadata);
+    super.visitNgComponent(metadata);
+  }
+
+  private validateComponent(metadata: ComponentMetadata): void {
+    if (metadata.selector) return;
+
+    const failure = getFailureMessage(metadata.controller.name!.text);
```

Comment:
We're very confident here. Do we know for sure that `name` is not going to be `null`?

Reply:
Hmm, to be sincere, I really don't know if we can receive a `null` name here. It was copied from `enforceComponentSelector`, anyway, I'll add a check for this :)